web/src/utils/ellipsis.ts

```diff
@@ -0,0 +1,656 @@
+/* eslint-disable */
+// @ts-nocheck
+
+/*!
+ *	dotdotdot JS 4.1.0
+ *
+ *	dotdotdot.frebsite.nl
+ *
+ *	Copyright (c) Fred Heusschen
+ *	www.frebsite.nl
+ *
+ *	License: CC-BY-NC-4.0
```

Comment:
The copyright license for this code is more restrictive than the MIT license memos uses particularly the noncommercial part.